### PR TITLE
Floating Menu Return Focus to Element

### LINKED ISSUE #11003: Floating Menu Return Focus to Element
## Feature Description

When a floating menu is active and user has entered it, focus should return to the element and allow the user to 'tab' to the next element when the user hits `esc` from the floating menu. 

## Acceptance Criteria 

- Hitting `esc` from floating menu returns focus to active element, tabbing again selects the next element
- Karma test for the above

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
Part of canvas keyboard nav work.

## Summary
Makes focus groups a little more stable, to where if you click into them, they behave the same as if you keyboard nav into them keybinding wise.

Also fixes exiting in/out of focus groups properly between floating menu -> element -> canvas focus wrapper.

## Relevant Technical Choices
Decided to have all focus groups rendered all the time. This works, because the only way to get into them is to manually focus (via click) or call one of the `EditLayerFocusManager` actions. This simplifies things and adds consistent behavior.

Also got rid of the concept of an `active focus group`. They're all always active now, and you can just navigate in/out of them with mouse or keybindings.

Added group specific cleanup functions since they're relatively static for the lifetime of the components. This fixes the focus order:
- in floating menu focus group
- press esc
- in element selection focus group
- press esc
- focused on canvas focus wrapper

## To-do
- [x] add some test coverage of new functionality

## User-facing changes
Improved consistency of interactions between mouse & keyboard nav with canvas focus groups

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. focus the sidepanel
2. click on an element
3 press `ctrl+alt+p` to keyboard nav into the floating menu
4. press `esc`
5 see that selected element has focus again
6. press `tab` or `shift/tab` a bunch and see that you're only navigating between elements
7. press `esc`
8. See that your focused on the canvas wrapper

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11003

### CODE REVIEW

**Review by BrookeGraham — DISMISSED** [review]
Works great. I think it is a lot easier to follow too.

**Comment by BrookeGraham** [comment]
Oh, @BrittanyIRL you were gonna add tests here too, right? I will revoke my ✅ Sorry for jumping the gun. I was excited.

File: packages/story-editor/src/components/canvas/karma/canvasKeyboardNavigation.karma.js [hunk 1]
```diff
@@ -67,6 +67,27 @@ describe('Canvas - keyboard navigation', () => {
     return document.activeElement.getAttribute('data-element-id');
   }
 
+  async function addElementsToCanvas() {
```
Comment (BrookeGraham) [inline packages/story-editor/src/components/canvas/karma/canvasKeyboardNavigation.karma.js:70]: Nice.

**Review by BrookeGraham — DISMISSED** [review]

**Review by BrittanyIRL — DISMISSED** [review]

**Review by BrittanyIRL — APPROVED** [review]

**Comment by BrittanyIRL** [comment]
I worked on this, so someone else should also look at it :)

**Review by BrookeGraham — APPROVED** [review]

**Comment by felipebochehin87** [comment]
@BrittanyIRL @littlemilkstudio 

I see the focus in two different elements at the same time, is that expected? Please see video attached.

1. Add an element to the canvas
2. Click on the element inside the canvas
3. Hit Esc - focus returns to the canvas, but I can still use the arrow keys to move the element that is inside the canvas

https://images.zenhubusercontent.com/235435637/9a6098b4-591f-47ad-bd26-a7b539599e06/2022_04_22_20_13_30.mp4

**Comment by felipebochehin87** [comment]
Thanks @BrittanyIRL! Moving this to QA Approved. :)

**Pull request merged**